### PR TITLE
encore test: fix conflict with pre-installed go versions

### PR DESCRIPTION
**Pull request by fredr**
In some corner case where a package didnt have any tests, running `encore test ./... -cover` caused errors like:
```
compile: version "go1.25.4-encore X:cacheprog" does not match go tool version "go1.25.4"
```

this is due a version conflict with a pre-installed go version. This makes sure we set the encore-go bin as the first item in the PATH so that wont happen

### CODE REVIEW

File: v2/compiler/build/tests.go [hunk 1]
```diff
@@ -188,6 +189,11 @@ func (b *builder) generateTestSpec(testCfg *GenerateTestSpecConfig) *TestSpec {
 		env = append(env, "CGO_ENABLED=0")
 	}
 
+	// prefix PATH with encore-go, so it doesnt conflict with other installed go versions
+	// if not set causes problems when running cover tests in go
+	path := goroot.Join("bin").ToIO() + string(filepath.ListSeparator) + os.Getenv("PATH")
```
Comment (eandre) [inline v2/compiler/build/tests.go]: We should probably prefer using `PATH` from `b.cfg.Env` if it is provided there, by looping through the list looking for `PATH=` as a prefix (can use `strings.CutPrefix`). And falling back to os.Getenv("PATH") otherwise.

Comment (fredr) [reply v2/compiler/build/tests.go]: we do send in the whole os.Environ there it seems (maybe we shouldnt?), so then this fix wont help as that PATH will take prescience then.

Comment (fredr) [reply v2/compiler/build/tests.go]: ah sorry, now I understood what you meant!

Comment (fredr) [reply v2/compiler/build/tests.go]: fixed!

**Review by eandre — APPROVED** [review]
